feat: support mysql backend as keda trigger

claneys opened the pull request:
Closes: #36166

Airflow currently support two SQL database backend with postgresql and mysql. PostgreSQL being the default one it's also the trigger used by default by the Keda ScaledObject created when enabling autoscaling. Which do not work if using a mysql database. Query has to be changed a bit and [mysql trigger parameters](https://keda.sh/docs/2.12/scalers/mysql/#authentication-parameters) aren't the same (pretty sneaky little changes within key names) and most of all the connection string used isn't the correct one when addressing to mysql.

claneys committed:
feat: support mysql backend as keda trigger

#36166

Signed-off-by: Romain Forlot <romain.forlot@dailymotion.com>

hussein-awala commented on chart/templates/_helpers.yaml:
```diff
@@ -84,7 +84,7 @@ If release name contains chart name it will be used as a full name.
         key: connection
   {{- end }}
-  {{- if and .Values.workers.keda.enabled .Values.pgbouncer.enabled (not .Values.workers.keda.usePgbouncer) }}
+  {{- if (or (and .Values.workers.keda.enabled (eq .Values.data.metadataConnection.protocol "mysql")) (and .Values.workers.keda.enabled .Values.pgbouncer.enabled (not .Values.workers.keda.usePgbouncer))) }}
```
This could be simplified to:
```suggestion
  {{- if and .Values.workers.keda.enabled (or (eq .Values.data.metadataConnection.protocol "mysql") (and .Values.pgbouncer.enabled (not .Values.workers.keda.usePgbouncer))) }}
```

claneys replied:
Thanks :)

hussein-awala reviewed (commented):
Could you add tests for mounted secret, kedaConnection value, and KEDA scaler when the protocol is MySQL?

claneys committed:
chore: clearer condition test

Thanks @hussein-awala for review

Signed-off-by: Romain Forlot <romain.forlot@dailymotion.com>

claneys committed:
test: Add helm-test for mysql keda support

Testing that required key are present for the trigger definition and at the right value

Signed-off-by: Romain Forlot <romain.forlot@dailymotion.com>

claneys committed:
ci: update format using ruff to pass static checks

Signed-off-by: Romain Forlot <romain.forlot@dailymotion.com>

claneys committed:
ci: remove unused python module

Signed-off-by: Romain Forlot <romain.forlot@dailymotion.com>
